Tasks/ServiceFabricUpdateAppVersions/Find-FileChanges.psm1

```diff
@@ -0,0 +1,114 @@
+﻿function Get-PackageFiles ($Path)
+{
+    # List all files except '.pdb' files because their content changes every build even with the 'deterministic' compiler flag
+    Find-VstsFiles -LiteralDirectory $Path -LegacyPattern "**;-:**\*.pdb" -Force |
+        ForEach-Object { $_.ToLower() } |
+        Sort-Object
+}
+
+function Find-FileChanges
+{
+    [CmdletBinding()]
+    [OutputType([bool])]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path $_})]
+        [string]
+        $NewPackageRoot,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path $_})]
+        [string]
+        $OldPackageRoot,
+
+        [string]
+        $LogIndent,
+
+        [Switch]
+        $LogAllChanges
+    )
```

Comment:
fyi - You may find the trace-vstsenteringinvocation function helpful for functions like this when diagnosing issues from the logs. It walks the psboundparameter hashtable and traces each entry. When customers run into issues, they can simply set system.debug to true, run the build again, and send the logs. If the trace entering methods are sprinkled throughout, it adds a significant amount of tracing almost for free.